tinytga/src/lib.rs

```diff
@@ -7,65 +7,97 @@
+//! let tga: Tga<Rgb888> = Tga::from_slice(data).unwrap();
 //!
-//! // Check the size of the image.
-//! assert_eq!(img.size(), Size::new(4, 4));
+//! let image = Image::new(&tga, Point::zero());
 //!
-//! // Collect pixels into a vector.
-//! let pixels: Vec<_> = img.pixels().collect();
+//! image.draw(&mut display)?;
+//! # Ok::<(), core::convert::Infallible>(()) }
 //! ```
 //!
-//! ## Drawing an image using `embedded-graphics`
+//! ## Using `DynamicTga` to draw an image
 //!
-//! This example demonstrates how a TGA image can be drawn to a [embedded-graphics] draw target.
+//! The previous example had the limitation that the color format needed to be known at compile
+//! time. In some use cases this can be a problem, for example if user supplied images should
+//! be displayed. To handle these cases [`DynamicTga`] can be used, which performs color conversion
+//! if necessary.
 //!
 //! ```rust
 //! # fn main() -> Result<(), core::convert::Infallible> {
-//! # let mut display = embedded_graphics::mock_display::MockDisplay::default();
+//! # let mut display = embedded_graphics::mock_display::MockDisplay::<Rgb888>::default();
 //! use embedded_graphics::{image::Image, pixelcolor::Rgb888, prelude::*};
-//! use tinytga::Tga;
+//! use tinytga::DynamicTga;
 //!
 //! // Include an image from a local path as bytes
 //! let data = include_bytes!("../tests/chessboard_4px_rle.tga");
-
-//! let tga: Tga<Rgb888> = Tga::from_slice(data).unwrap();
+//!
+//! let tga = DynamicTga::from_slice(data).unwrap();
 //!
 //! let image = Image::new(&tga, Point::zero());
 //!
 //! image.draw(&mut display)?;
 //! # Ok::<(), core::convert::Infallible>(()) }
 //! ```
+//! ## Accessing pixels using an embedded-graphics color type
+//!
+//! Even if tinytga is used without using [embedded-graphics] to draw the image the color types
+//! provided by [embedded-graphics] can still be used to access the pixel data using the
+//! [`pixels`](struct.Tga.html#method.pixels) method.
+//!
+//! ```rust
+//! use embedded_graphics::{prelude::*, pixelcolor::Rgb888};
+//! use tinytga::{Bpp, ImageOrigin, ImageType, RawPixel, Tga, TgaHeader};
+//!
+//! // Include an image from a local path as bytes
+//! let data = include_bytes!("../tests/chessboard_4px_rle.tga");
+//!
+//! // Create a TGA instance from a byte slice.
+//! // The color type is set by defining the type of the `img` variable.
+//! let img: Tga<Rgb888> = Tga::from_slice(data).unwrap();
+//!
+//! // Check the size of the image.
+//! assert_eq!(img.size(), Size::new(4, 4));
+//!
+//! // Collect pixels into a vector.
+//! let pixels: Vec<_> = img.pixels().collect();
+//! ```
 //!
 //! ## Accessing raw pixel data
 //!
 //! If you do not want to use the color types provided by [embedded-graphics] you can also access
-//! the raw image data.
+//! the raw image data. The iterator returned by the [`pixels`](struct.RawTga.html#method.pixels)
+//! method uses `u32` values to return the raw color value of each pixel.
```

Comment:
This paragraph might be better written as something like:

```
//! If [embedded-graphics] is not used in the target application, the raw image data can be accessed
//! with the [`pixels`](struct.RawTga.html#method.pixels) method on [`RawTga`](struct.RawTga.html). 
//! The returned iterator produces a `u32` for each pixel value.
```

The "you" wording sounds a bit odd compared to the rest of the docs that use an imperative form.

Reply:
Thanks for the suggestion, I'll use that.